app/models/setting/discovered.rb

```diff
@@ -19,12 +19,19 @@ def self.load_defaults
         self.set('discovery_fact', N_("Fact name to use for primary interface detection"), "discovery_bootif"),
         self.set('discovery_hostname', N_("List of facts to use for the hostname (separated by comma, first wins)"), "discovery_bootif"),
         self.set('discovery_auto', N_("Automatically provision newly discovered hosts, according to the provisioning rules"), false),
+        self.set('discovery_lock', N_("Automatically generate PXE configuration to pin a newly discovered host to discovery"), false),
         self.set('discovery_reboot', N_("Automatically reboot discovered host during provisioning"), true),
       ].compact.each { |s| self.create s.update(:category => "Setting::Discovered")}
     end
 
     Setting.transaction do
       [
+          self.set('discovery_lock_template', N_("PXE template to be used when pinning a host to discovery"), ''),
```

Comment:
Give the snippet a reasonable name (e.g. "pxelinux_discovery") and use it as the default value here.